src/oc/web/actions/activity.cljs

```diff
@@ -640,7 +644,15 @@
         (when (or (= change-type :add)
                   (= change-type :delete))
           (when (= (router/current-board-slug) "all-posts")
-            (all-posts-get (dis/org-data) (dis/ap-initial-at)))
+            (all-posts-get (dis/org-data) (dis/ap-initial-at)
+             ;; In case another user is adding a new post
+             ;; mark it as unread directly
+             (when (and (= change-type :add)
+                        (not= (:user-id data) (jwt/user-id)))
+               (fn [{:keys [success]}]
+                 (when success
+                   (dis/dispatch! [:mark-unread (router/current-org-slug) {:uuid activity-uuid
+                                                                           :board-uuid section-uuid}]))))))
```

Comment:
@bago2k4 I'm having trouble understanding this section of code. Why is it necessary to explicitly mark a new post as unread when it comes from the change service? Shouldn't a new post already default to the unread state? I'm not so much criticizing the code as much as I am trying to understand it.

Reply:
Good question indeed.
So the change service doesn't say what posts are read, instead it says which one are unread. Not sure this is the optimal way to do this but it was how the seen was working and we decided to use the same logic for read (for dev time and also logic complexity).
As a result of that if a new post is added from the current session that post will be in read state until Change Service receive the message (SQS message) that a new post has been added and push to the client that as an unread.
To avoid any possible delay introduced by SQS or WS connection I make sure the newly published post is marked as unread, only in the current local state though.
Make sense @cjsauer ?

Reply:
@bago2k4 yes that makes sense, thanks for the explanation

Reply:
@cjsauer yeah, i was actually referring to another situation where the post is added from the current user.
You were asking about a situation where the post is being added from another user and the client receives it from the CS connection.
But the logic is still the same, the post is marked directly as unread to avoid delays btw the new post add and the refresh of the list of unreads posts.